src/Editor/GUI/Editors/AnimationControllerEditor.cpp

```diff
@@ -0,0 +1,109 @@
+#include <Utility/Log.hpp>
+#include <Engine/Manager/Managers.hpp>
+#include <Engine/Manager/ResourceManager.hpp>
+#include <Engine/Hymn.hpp>
+
+using namespace GUI;
+
+AnimationControllerEditor::AnimationControllerEditor() {
+}
+
+AnimationControllerEditor::~AnimationControllerEditor() {
+}
+
+void AnimationControllerEditor::SetAnimationController(Animation::AnimationController * animationController) {
+    if (this->animationController != nullptr)
+        this->animationController->Save(Hymn().GetPath() + "/" + animationController->path + animationController->name + ".asset");
+
+    this->animationController = animationController;
+}
+
+Animation::AnimationController * AnimationControllerEditor::GetAnimationController() {
+    return animationController;
+}
+
+void AnimationControllerEditor::SetVisible(bool visible) {
+    NodeEditor::SetVisible(visible);
+    if (this->animationController != nullptr)
+        animationController->Save(Hymn().GetPath() + "/" + animationController->path + animationController->name + ".asset");
+}
+
+void AnimationControllerEditor::ShowContextMenu() {
+    if (animationController == nullptr) {
+        Log() << "ERROR: No animation controller choosen.\n";
+        return;
+    }
+
+    if (ImGui::MenuItem("Add animation action")) {
+        Animation::AnimationController::AnimationAction* newAction = new Animation::AnimationController::AnimationAction;
+        std::string name = "Action #" + std::to_string(animationController->animationNodes.size() + 1);
+        memcpy(newAction->name, name.c_str(), name.size() + 1);
+        newAction->index = animationController->animationNodes.size();
+        animationController->animationNodes.push_back(newAction);
+    }
+
+    if (ImGui::MenuItem("Add animation transition")) {
+        Animation::AnimationController::AnimationTransition* newTransition = new Animation::AnimationController::AnimationTransition;
+        std::string name = "Animation transition: " + std::to_string(animationController->animationNodes.size() + 1);
+        memcpy(newTransition->name, name.c_str(), name.size() + 1);
+        animationController->animationNodes.push_back(newTransition);
+    }
+}
+
+void AnimationControllerEditor::ShowNode(Node * node) {
+    ImGui::Text("Action: %s", node->name);
+    ImGui::InputText("Name", node->name, 128);
+
+    if (typeid(*node) == typeid(Animation::AnimationController::AnimationAction)) {
+        Animation::AnimationController::AnimationAction * action = dynamic_cast<Animation::AnimationController::AnimationAction*>(node);
+        if (ImGui::Button("Select animation clip##Clip"))
+            ImGui::OpenPopup("Select animation clip##Clip");
+
+        if (ImGui::BeginPopup("Select animation clip##Clip")) {
+            ImGui::Text("Select animation");
+            ImGui::Separator();
+
+            if (resourceSelector.Show(ResourceList::Resource::Type::ANIMATION_CLIP)) {
+                if (action->animationClipName != nullptr)
```

Comment:
`action` is the result of a `dynamic_cast` so it should be checked for validity before dereferencing it.